source/SemanticModel.cpp

```diff
@@ -69,13 +71,65 @@ InstanceSymbol* SemanticModel::makeImplicitInstance(const ModuleDeclarationSynta
     return alloc.emplace<InstanceSymbol>(module, true);
 }
 
+void SemanticModel::makePackages() {
+    HashMap<StringRef, SourceLocation> nameDupMap;
+    for (auto pkg : declTable.getPackages()) {
+        auto name = pkg->header->name.valueText();
+        Scope *scope = alloc.emplace<Scope>();
+        ASSERT(packages.add(alloc.emplace<ModuleSymbol>(pkg, scope, ArrayRef<const Symbol *>())));
```

Comment:
Don't put code that must execute inside an ASSERT()... if it gets compiled out (which ASSERTs often due in optimized builds) then your code won't run anymore.